### ISSUE TITLE
Feature request: "Hide a comment" on Pull Requests and Issues

### ISSUE THREAD

**Issue by BhaaLseN** [go-gitea/gitea#5262]
I was happy to see Gitea had a "Review" feature on Pull Requests similar to GitHub when I was trialling it as replacement for our current hosting, but I'm kinda-of missing the ["Hide a comment"](https://help.github.com/articles/managing-disruptive-comments/#hiding-a-comment) feature.

Our main use-case for this would be hiding both PR comments and Code comments that no longer apply (either as "Resolved" or "Outdated"), but it wouldn't hurt to include other options for a more complete experience (such as "Off-topic", "Spam" or "Abuse" for example; but need not limited to those reasons).

We got an external issue tracker, but I'm certain it would also make sense with the built-in tracker to apply to Issues.

Comments would be displayed similar to how GitHub handles it, allowing a reader to expand them if needed:
![hidden-comment](https://user-images.githubusercontent.com/826052/47952039-89b99080-df69-11e8-8ecf-a61667c9da53.png)

**Comment by 6543** [comment]
Comment already has a `ResolveDoerID` field.

We just have to add a ResolveReason witch could be a int (enum) or varchar if answers can be customized ...

TODOs:
 * [ ] extend struct [Go]
 * [ ] add action (similar to edit) for hide/unhide [Go]
 * [ ] update template to get menue and hide if set [HTML]
 * [ ] add smal migration based on struct update [ Go]

**Comment by segfult** [comment]
hey @6543, can i have a knock at this?

**Comment by 6543** [comment]
@segfult sure go on ! (ps: target will be v1.15.0 then so dont rush :) )

**Comment by jpjamipark** [comment]
Hi @6543 , I'd like to work on this.

Just wanted to clarify, it looks like currently only PR comments are resolve-able, but issue comments are not. 

Would it cause problems to also make issue comments hide-able/resolve-able?

**Comment by 6543** [comment]
yes this is a feature request for issue and pull, its about the "normal comments" :)

and you can follow the hits I describe above to get the job done :)
